test/e2e/framework/framework.go

```diff
@@ -0,0 +1,268 @@
+package framework
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pmezard/go-difflib/difflib"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func init() {
+	rand.Seed(time.Now().Unix())
+}
+
+// CompareWithFixtureDir will compare all files in a directory with a corresponding test fixture directory.
+func CompareWithFixtureDir(t *testing.T, golden, output string) {
```

Comment:
These are not specific to e2e tests, so can we keep them where they are?

Mainly asking because we want the actual e2e tests to be gated by a buildtag but we do not want these helpers to be gated by anything, as they are used in unittests

Reply:
These are unrelated to what is used in unit